gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -446,6 +443,24 @@ && canUseDirectPathWithUniverseDomain()) {
     return managedChannel;
   }
 
+  /**
+   * Marked as Internal Api and intended for internal use. DirectPath must be enabled via the
+   * settings and a few other configurations/settings must also be valid for the request to go
+   * through DirectPath.
+   *
+   * <p>Checks: 1. Credentials are compatible 2.Running on Compute Engine 3. Universe Domain is
+   * configured to for the Google Default Universe
+   *
+   * @return if DirectPath is enabled for the client AND if the configurations are valid
+   */
+  @InternalApi
+  public boolean canUseDirectPath() {
```

Comment:
How do we imagine Spanner use this getter? I think it might be better for us to create a way in `MetricTracer` to take common client level attributes, otherwise they have to pass this value around to each RPC calls.

Reply:
Discussed offline. Adding a new map to be able to pass client level attributes to the MetricsTracerFactory.